### PR TITLE
mrc-1231: nicer interface for detecting crashed workers

### PR DESCRIPTION
**Pull request by richfitz**
This PR adds a public method for detecting crashed workers, and tidies up the embryonic code for supporting it

### CODE REVIEW

File: tests/testthat/test-zzz-fault-tolerance.R [hunk 1]
```diff
@@ -237,3 +187,104 @@ test_that("race conditioning handling when cancelling tasks", {
...
+
+  ## This is a bit annoying as it takes a while to run through;
+  Sys.sleep(expire)
+
+  ## Our key has gone!  Marvellous!
+  expect_equal(obj$con$EXISTS(key), 0)
+
+  expect_equal(obj$worker_list(), wid)
+  msg <- capture_messages(res <- obj$worker_detect_exited())
+  expect_equal(res, set_names(t, wid))
```
Comment (r-ash) [inline tests/testthat/test-zzz-fault-tolerance.R]: Should we move this check to after `res` is created?

File: R/rrq_controller.R [hunk 2]
```diff
@@ -158,6 +158,7 @@ R6_rrq_controller <- R6::R6Class(
       worker_list(self$con, self$keys)
     },
 
+    ## Lists workers *known* to have exited
     worker_list_exited = function() {
```
Comment (r-ash) [inline R/rrq_controller.R]: What conditions causes a worker to know it has exited?

File: tests/testthat/test-zzz-fault-tolerance.R (hunk 1 above)
Comment (richfitz) [reply tests/testthat/test-zzz-fault-tolerance.R]: res is created on line 281 (inside the `capture_messages`)

File: R/rrq_controller.R (hunk 2 above)
Comment (richfitz) [reply R/rrq_controller.R]: Any clean exit, and any workers that have been detected in a previous round of worker_detect_exited

File: tests/testthat/test-zzz-fault-tolerance.R (hunk 1 above)
Comment (r-ash) [reply tests/testthat/test-zzz-fault-tolerance.R]: AH sorry - misread that

**Pull request merged**